### PR TITLE
[Horovod-Runtime] Introduce custom horovod driver script in debug mode

### PR DESCRIPTION
**Pull request by zuston**

### CODE REVIEW

File: tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java [hunk 1]
```diff
@@ -305,4 +305,8 @@ public static String getContainerDockerMountKey() {
   public static final boolean DEFAULT_TEST_HOROVOD_FAIL = false;
   public static final String IN_TEST_HOROVOD_MODE = TONY_APPLICATION_PREFIX + "test.horovod-test-mode-enable";
   public static final boolean DEFAULT_IN_TEST_HOROVOD_MODE = false;
+
+  // For Horovod
+  public static final String HOROVOD_DRIVER_DEBUG_MODE_ENABLE = TONY_APPLICATION_PREFIX + "horovod.driver-debug-mode-enable";
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java:310]: `tony.horovod.driver.mode.debug`

File: tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java [hunk 2]
```diff
@@ -305,4 +305,8 @@ public static String getContainerDockerMountKey() {
   public static final boolean DEFAULT_TEST_HOROVOD_FAIL = false;
   public static final String IN_TEST_HOROVOD_MODE = TONY_APPLICATION_PREFIX + "test.horovod-test-mode-enable";
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java:306]: `tony.horovod.mode.test`

**Review by oliverhu — COMMENTED** [review]
add some documentation on how to use this mode?

File: tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java [hunk 3]
```diff
@@ -301,12 +301,12 @@ public static String getContainerDockerMountKey() {
       Arrays.asList(CONTAINER_LAUNCH_ENV, EXECUTION_ENV, getContainerResourcesKey()));
 
   // Local testing horovod driver
-  public static final String TEST_HOROVOD_FAIL_ENABLE_KEY = TONY_APPLICATION_PREFIX + "test.horovod-driver-fail-enable";
+  public static final String TEST_HOROVOD_FAIL_ENABLE_KEY = TONY_PREFIX + "horovod.mode.test.fast.fail";
```
Comment (oliverhu) [inline tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java:304]: create a KEY for `tony_prefx.horovod` ?

**Review by oliverhu — APPROVED** [review]

File: tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java (hunk 3 above)
Comment (zuston) [reply tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java:304]: Make sense.

**Pull request merged**